hbase-common/src/main/java/org/apache/hadoop/hbase/ChoreService.java

```diff
@@ -352,17 +349,14 @@ public boolean isTerminated() {
   }
 
   private void cancelAllChores(final boolean mayInterruptIfRunning) {
-    ArrayList<ScheduledChore> choresToCancel = new ArrayList<>(scheduledChores.keySet().size());
     // Build list of chores to cancel so we can iterate through a set that won't change
     // as chores are cancelled. If we tried to cancel each chore while iterating through
     // keySet the results would be undefined because the keySet would be changing
-    for (ScheduledChore chore : scheduledChores.keySet()) {
-      choresToCancel.add(chore);
-    }
+    ArrayList<ScheduledChore> choresToCancel = new ArrayList<>(scheduledChores.keySet());
+
     for (ScheduledChore chore : choresToCancel) {
       cancelChore(chore, mayInterruptIfRunning);
     }
-    choresToCancel.clear();
```

Comment:
No need to `clear` this data structure.  All entries will be cleaned up once the `ArrayList` itself is.